device.c

```diff
@@ -158,18 +165,54 @@ static int vcam_try_fmt_vid_cap(struct file *file,
             vcam_sizes, n_avail, width, height, vcam_sizes[n_avail - 1].width,
             vcam_sizes[n_avail - 1].height);
 #else
-        const struct v4l2_discrete_probe vcam_probe = {
-            vcam_sizes, ARRAY_SIZE(vcam_sizes)};
+        const struct v4l2_discrete_probe vcam_probe = {vcam_sizes, n_avail};
 
         const struct v4l2_frmsize_discrete *sz =
             v4l2_find_nearest_format(&vcam_probe, vcam_sizes[n_avail - 1].width,
                                      vcam_sizes[n_avail - 1].height);
 #endif
         f->fmt.pix.width = sz->width;
         f->fmt.pix.height = sz->height;
+        dev->output_format.width = sz->width;
+        dev->output_format.height = sz->height;
+        if (dev->output_format.pixelformat == V4L2_PIX_FMT_YUYV) {
```

Comment:
Add some comments about the pixel format and/or color space such as `V4L2_COLORSPACE_SMPTE170M`.